### PR TITLE
Oracle

### PR DESCRIPTION
**Pull request by anthony-apeark**

### CODE REVIEW

File: contracts/src/PriceFeeds/PythPriceFeed.sol [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    function fetchRedemptionPrice() external returns (uint256, bool) {
+        return _fetchPrice();
+    }
+
+    function lastGoodPrice() external view returns (uint256) {
+        return _lastGoodPrice;
+    }
+
+    function _fetchPrice() internal returns (uint256, bool) {
+        PythStructs.Price memory price = pythContract.getPriceNoOlderThan(
```
Comment (jnosleep) [inline contracts/src/PriceFeeds/PythPriceFeed.sol:48]: need to try catch this like ChainlinkPriceFeed
ref for getPriceNoOlderThan implementation: https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/solidity/AbstractPyth.sol#L57

File: contracts/src/PriceFeeds/PythPriceFeed.sol [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        borrowerOperations = IBorrowerOperations(_borrowOperationsAddress);
+
+        _renounceOwnership();
+    }
+
+    function _scalePriceTo18decimals(
+        int256 _price,
+        int32 _decimals
+    ) internal pure returns (uint256) {
+        return uint256(_price) * 10 ** uint256(18 + int256(_decimals));
```
[the remark this answers is not in the document]
Comment (jnosleep) [reply contracts/src/PriceFeeds/PythPriceFeed.sol:67]: maybe can use this util?
https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/solidity/PythUtils.sol#L13

[the remark this answers is not in the document]
Comment (jnosleep) [reply contracts/src/PriceFeeds/PythPriceFeed.sol:67]: example: https://github.com/pyth-network/pyth-examples/blob/main/price_feeds/evm/oracle_swap/contract/src/OracleSwap.sol#L81

[the remark this answers is not in the document]
Comment (anthony-apeark) [reply contracts/src/PriceFeeds/PythPriceFeed.sol]: expo in pyth is different from chainlink, so should be add.
Migrated to PythUtils

File: contracts/src/test/OracleMainnet.t.sol [hunk 3]
```diff
@@ -263,6 +284,38 @@ contract OraclesMainnet is TestAccounts {
...
+        assertEq(fetchedLdoUsdPrice, expectedPrice);
+    }
+
+    function testFetchPriceReturnsCorrectPriceLqty() public {
+        (uint256 fetchedLqtyUsdPrice,) = lqtyPriceFeed.fetchPrice();
+        assertGt(fetchedLqtyUsdPrice, 0);
+
+        PythStructs.Price memory price = pythContract.getPriceUnsafe(0x5e8b35b0da37ede980d8f4ddaa7988af73d8c3d110e3eddd2a56977beb839b63);
+
+        uint256 expectedPrice = uint256(int256(price.price)) * 10 ** uint256(18 + int256(price.expo));
```
Comment (jnosleep) [inline contracts/src/test/OracleMainnet.t.sol:314]: this test still works?

Comment (anthony-apeark) [reply contracts/src/test/OracleMainnet.t.sol:314]: yes

**Review by jnosleep — APPROVED** [review]

**Pull request merged**